flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/InputProcessorUtil.java

```diff
@@ -79,11 +80,26 @@ public static CheckpointedInputGate createCheckpointedInputGate(
 			unionedInputGates[i] = InputGateUtil.createInputGate(inputGates[i].toArray(new IndexedInputGate[0]));
 		}
 
+		IntStream numberOfInputChannelsPerGate =
+			Arrays
+				.stream(inputGates)
+				.flatMap(collection -> collection.stream())
+				.sorted(Comparator.comparingInt(IndexedInputGate::getGateIndex))
+				.mapToInt(InputGate::getNumberOfInputChannels);
+
 		Map<InputGate, Integer> inputGateToChannelIndexOffset = generateInputGateToChannelIndexOffsetMap(unionedInputGates);
+		// Note that numberOfInputChannelsPerGate and inputGateToChannelIndexOffset have a bit different
```

Comment:
Yes... It took me a while to understand their usage. Why don't we make them consistent by easily sorting the `inputGates` before doing anything here (there's no risk if I understand correctly), so that it won't bother us in the future.

Reply:
Hmmm, I'm not sure, as what if left input has input gates with indexes `0` and `3`, while the right input has indexes `1`, `2` and `4`? (I'm not sure if that's a valid scenario in the JobGraphGenerator) Left input would have a one instance of `UnionInputGate` over gates 0 and 3, while right input would have another instance with gates 1, 2 and 4. However we sort them, it would be somehow inconsistent?

Reply:
You're right. I didn't notice that `inputGateToChannelIndexOffset`'s key is an unioned InputGate. Thanks for pointing this out.